flybywiresim/a32nx
# feat(mcdu): Messages dynamic field compatibility

Fixes #3681

## Goal of this PR
Allow a temporary edit of a message (and maintain "thread safety") to allow messages that have a dynamic field attached to it to be editable.
Basically create support for messages like:
- `AREA RNP IS XX.XX`
- `NEW ACC ALT-HHHH `
- `NEW CRZ ALT-HHHHH`
- `NEW THR RED ALT-HHHH`
- `PROCEDURE RNP IS XX.XX`
- `USING COST INDEX--NNN`
- `XXXX IS DESELECTED`

## Summary of Changes

- Added McduMessage Class as preset
- Ported messages to new message class
- Added ability to "temporary" edit a message but remain thread safe.















Discord username (if different from GitHub): DerL30N#3751

## Testing instructions

Nothing really to test.
1. Check if "normal" messages appear as normal.
2. Climb to any specific cruise altitude change the cruise altitude -> observe message ![image](https://user-images.githubusercontent.com/1857322/108735372-c3df3f80-7530-11eb-9dac-ba77dcb6a56f.png)
3. Repeat 2. at least once



## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link
